test/chemistry/test_vqe_uccsd_adapt.py

```diff
@@ -79,13 +79,36 @@ def test_vqe_adapt(self):
                                    self.num_particles, initial_state=self.init_state)
         backend = Aer.get_backend('statevector_simulator')
         optimizer = L_BFGS_B()
+        algorithm = VQEAdapt(self.qubit_op, self.var_form_base, optimizer,
+                             threshold=0.00001, delta=0.1, max_iterations=1)
+        result = algorithm.run(backend)
+        self.assertEqual(result.num_iterations, 1)
+        self.assertEqual(result.finishing_criterion, 'Maximum number of iterations reached')
+
         algorithm = VQEAdapt(self.qubit_op, self.var_form_base, optimizer,
                              threshold=0.00001, delta=0.1)
         result = algorithm.run(backend)
         self.assertAlmostEqual(result.eigenvalue.real, -1.85727503, places=2)
-        self.assertIsNotNone(result.num_iterations)
-        self.assertIsNotNone(result.final_max_gradient)
-        self.assertIsNotNone(result.finishing_criterion)
+        self.assertEqual(result.num_iterations, 2)
+        self.assertAlmostEqual(result.final_max_gradient, 0.0, places=5)
+        self.assertEqual(result.finishing_criterion, 'Threshold converged')
+
+    def test_vqe_adapt_check_cyclicity(self):
+        """ VQEAdapt index cycle detection """
+        param_list = [
+            ([1, 1], True),
+            ([1, 2, 1], False),
+            ([1, 2, 1, 1], True),
+            ([1, 2, 1, 2], True),
+            ([1, 12], False),
+            ([1, 12, 2], False),
+            ([1, 11, 1, 111], False),
+            ([1, 2, 3, 1, 2, 3], True),
+            ([1, 2, 3, 4, 1, 2, 3, 4], True),
+        ]
```

Comment:
Sorry, but I found another counterexample: [11 2 1 2]

Reply:
Well yeh I guess I should have seen that one coming. I already have the fix but this time I will ensure by test set is complete before pushing again.